### PR TITLE
A few small conversion fixes

### LINKED ISSUE #479: Multi-allelic split code may fail when accessing PL
When splitting a multi-allelic variant from a VCF, the VCF record may be tagged with the PLs, but not have _all_ of the PLs attached. E.g.:

```
fnothaft$ ./bin/adam-submit vcf2adam ALL.chrY.phase3_integrated.20130502.genotypes.vcf chrY.adam
Spark assembly has been built with Hive, including Datanucleus jars on classpath
2014-11-14 17:07:48.306 java[50061:1903] Unable to load realm info from SCDynamicStore
2014-11-14 17:07:53 WARN  NativeCodeLoader:52 - Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
[HG00096 T* GQ 99 PL 0,1238,68 {CN=1, CNL=-123.78,-0,-6.77, CNP=-125.44,-0,-9.85, CNQ=98.5, GP=-0,-125.44,-9.85}] / 2
true
[0, 1238, 68]
[0, 3, 5]
2014-11-14 17:08:25 ERROR Executor:96 - Exception in task 0.0 in stage 1.0 (TID 6)
java.lang.ArrayIndexOutOfBoundsException: 3
	at org.bdgenomics.adam.converters.VariantContextConverter$$anonfun$convert$3$$anonfun$1.apply$mcII$sp(VariantContextConverter.scala:141)
	at org.bdgenomics.adam.converters.VariantContextConverter$$anonfun$convert$3$$anonfun$1.apply(VariantContextConverter.scala:141)
	at org.bdgenomics.adam.converters.VariantContextConverter$$anonfun$convert$3$$anonfun$1.apply(VariantContextConverter.scala:141)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofInt.foreach(ArrayOps.scala:156)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:244)
	at scala.collection.mutable.ArrayOps$ofInt.map(ArrayOps.scala:156)
```

### PR DESCRIPTION
**Pull request by fnothaft**
Resolves #479. Also adds a bit of logging to the legacy format converters.

### CODE REVIEW

**Comment by fnothaft** [comment]
Rebased.

File: adam-core/src/main/scala/org/bdgenomics/adam/converters/SAMRecordConverter.scala [hunk 1]
```diff
@@ -17,168 +17,187 @@
...
+            builder.setSecondaryAlignment(!samRecord.getSupplementaryAlignmentFlag)
+          }
         }
       }
-    }
 
-    // Position of the mate/next segment
-    val mateReference: Int = samRecord.getMateReferenceIndex
+      // Position of the mate/next segment
+      val mateReference: Int = samRecord.getMateReferenceIndex
```
Comment (massie) [inline adam-core/src/main/scala/org/bdgenomics/adam/converters/SAMRecordConverter.scala]: Is this indentation intentional? I'm surprised that Scalariform didn't complain.

File: adam-core/src/main/scala/org/bdgenomics/adam/converters/SAMRecordConverter.scala [hunk 2]
```diff
@@ -17,168 +17,187 @@
...
 
-    if (mateReference != SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX) {
-      builder.setMateContig(SequenceRecord.toADAMContig(dict(samRecord.getMateReferenceName).get))
+      if (mateReference != SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX) {
+        builder.setMateContig(SequenceRecord.toADAMContig(dict(samRecord.getMateReferenceName).get))
```
Comment (massie) [inline adam-core/src/main/scala/org/bdgenomics/adam/converters/SAMRecordConverter.scala]: Same here.

File: adam-core/src/main/scala/org/bdgenomics/adam/converters/SAMRecordConverter.scala (hunk 1 above)
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/converters/SAMRecordConverter.scala]: The patch modifies the conditionals surrounding this code block, which leads to an indentation change for this stretch of code.

**Comment by massie** [comment]
Thanks, Frank!

**Pull request merged**